ledger/common/encoding/encoding.go

```diff
@@ -505,6 +589,9 @@ func decodeTrieUpdate(inp []byte) (*ledger.TrieUpdate, error) {
 		return nil, fmt.Errorf("error decoding trie update: %w", err)
 	}
 
+	paths := make([]ledger.Path, numOfPaths)
+	payloads := make([]*ledger.Payload, numOfPaths)
```

Comment:
I think there is a miss-conception involved here. Up to this moment, I also had this (apparently incorrect) assumption:
* When allocating the underlying array for a slice, the compiler needs to write the default values to each slice element.  
* We were under the impression that if you initialize a slice with zero-length but non-zero capacity, the compiler would skip this initialization step.

But your example shows that this assumption is false. Thanks, I learned something 🎉 : 
* https://go.dev/blog/slices-intro explicitly states 
   > Arrays do not need to be initialized explicitly; the zero value of an array is a ready-to-use array whose elements are themselves zeroed:

Reply:
Thanks @fxamacker for clarifying this mistake! I mistakenly thought that the pattern :
```
t := make([]int, 0, 10)
for i:=0; i<10; i++ {
  t = append(t, val)
}
```
is faster than :
```
t := make([]int, 10)
for i:=0; i<10; i++ {
  t[i] = val
}
```
because it avoids the initialization of t. It turns out this is not true! 
I've learned something today!